### PR TITLE
HARP-8249 Changed calculation for visibily of text labels in spherical projection

### PR DESCRIPTION
**Pull request by MPursche**
Signed-off-by: Marcel Pursche <marcel.pursche@here.com>

Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

**Comment by MPursche** [comment]
Rebuild failed please.

**Comment by MPursche** [comment]
@heremaps-bot retest this please

File: @here/harp-mapview/lib/text/Placement.ts [hunk 1]
```diff
@@ -43,12 +44,12 @@ function checkViewDistance(
...
-    // Spherical projection
+    // For sphere projection: Filter labels that are close to the horizon
     tmpPosition.copy(textElement.position).normalize();
-    camera.getWorldDirection(tmpCameraDir);
+    camera.getWorldPosition(tmpCameraDir).normalize();
+    const alpha = Math.acos(tmpPosition.dot(tmpCameraDir));
 
-    // TODO: Revisit, why is this angle check needed and where does the constant -0.6 come from?
-    return tmpPosition.dot(tmpCameraDir) < -0.6 && textDistance <= maxViewDistance
+    return alpha < TEXT_ELEMENT_FALLOFF_ANGLE && textDistance <= maxViewDistance
```
Comment (ninok) [inline @here/harp-mapview/lib/text/Placement.ts]: Just compare with cosine of TEXT_ELEMENT_FALLOFF_ANGLE so you would not need to compute the acos every frame

Comment (ninok) [reply @here/harp-mapview/lib/text/Placement.ts]: Also the return value should be undefined if the label is too far away. You are returning a bool

**Pull request merged**